contracts/Markets/StandardMarketWithPriceLogger.sol

```diff
@@ -0,0 +1,117 @@
+            startDate = _startDate;
+        }
+    }
+
+    /// @dev Allows market creator to close the markets by transferring all remaining outcome tokens to the creator
+    function close()
+        public
+    {
+        endDate = now;
+        super.close();
+    }
+
+    /// @dev Allows to buy outcome tokens from market maker
+    /// @param outcomeTokenIndex Index of the outcome token to buy
+    /// @param outcomeTokenCount Amount of outcome tokens to buy
+    /// @param maxCost The maximum cost in collateral tokens to pay for outcome tokens
+    /// @return Cost in collateral tokens
+    function buy(uint8 outcomeTokenIndex, uint outcomeTokenCount, uint maxCost)
+        public
+        returns (uint cost)
+    {
+        logPrice();
+        cost = super.buy(outcomeTokenIndex, outcomeTokenCount, maxCost);
+    }
+
+    /// @dev Allows to sell outcome tokens to market maker
+    /// @param outcomeTokenIndex Index of the outcome token to sell
+    /// @param outcomeTokenCount Amount of outcome tokens to sell
+    /// @param minProfit The minimum profit in collateral tokens to earn for outcome tokens
+    /// @return Profit in collateral tokens
+    function sell(uint8 outcomeTokenIndex, uint outcomeTokenCount, uint minProfit)
+        public
+        returns (uint profit)
+    {
+        logPrice();
+        profit = super.sell(outcomeTokenIndex, outcomeTokenCount, minProfit);
+    }
+
+    /// @dev Buys all outcomes, then sells all shares of selected outcome which were bought, keeping
+    ///      shares of all other outcome tokens.
+    /// @param outcomeTokenIndex Index of the outcome token to short sell
+    /// @param outcomeTokenCount Amount of outcome tokens to short sell
+    /// @param minProfit The minimum profit in collateral tokens to earn for short sold outcome tokens
+    /// @return Cost to short sell outcome in collateral tokens
+    function shortSell(uint8 outcomeTokenIndex, uint outcomeTokenCount, uint minProfit)
+        public
+        returns (uint cost)
+    {
+        logPrice();
+        cost = super.shortSell(outcomeTokenIndex, outcomeTokenCount, minProfit);
+    }
+
+    /// @dev Calculates average price for long tokens based on price integral
+    /// @return Average price for long tokens
+    function getAvgPrice()
+        public
+        returns (uint)
+    {
+        return priceIntegral / (now - startDate);
+    }
+
+    /*
+     *  Private functions
+     */
+    /// @dev Adds price integral since the last trade to the total price integral and updates last
+    ///      trade timestamp
+    function logPrice()
+        private
+    {
+        if (now >= startDate) {
+            // Calculate price of long tokens
+            uint price = marketMaker.calcMarginalPrice(this, LONG);
+            if (lastTrade > 0)
+                priceIntegral += price * (now - lastTrade);
+            else
+                priceIntegral += price * (now - startDate);
+            lastTrade = now;
+        }
```

Comment:
endDate is set once the market is closed. Once the market is closed trading is not possible anymore and the logPrice function cannot be executed anymore. With the current implementation endDate is not used at all and should probably be removed.